server/sonar-server/src/main/java/org/sonar/server/computation/issue/TrackerRawInputFactory.java

```diff
@@ -113,19 +115,20 @@ private DefaultIssue toIssue(LineHashSequence lineHashSeq, ScannerReport.Issue r
       DefaultIssue issue = new DefaultIssue();
       init(issue);
       issue.setRuleKey(RuleKey.of(reportIssue.getRuleRepository(), reportIssue.getRuleKey()));
-      if (reportIssue.hasLine()) {
-        issue.setLine(reportIssue.getLine());
-        issue.setChecksum(lineHashSeq.getHashForLine(reportIssue.getLine()));
+      if (reportIssue.hasTextRange()) {
+        int startLine = reportIssue.getTextRange().getStartLine();
+        issue.setLine(startLine);
+        issue.setChecksum(lineHashSeq.getHashForLine(startLine));
       } else {
         issue.setChecksum("");
       }
-      if (reportIssue.hasMsg()) {
+      if (isNotEmpty(reportIssue.getMsg())) {
         issue.setMessage(reportIssue.getMsg());
       }
-      if (reportIssue.hasSeverity()) {
+      if (reportIssue.getSeverity() != Severity.UNSET_SEVERITY) {
         issue.setSeverity(reportIssue.getSeverity().name());
       }
-      if (reportIssue.hasGap()) {
+      if (reportIssue.getGap() != 0) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1244)